diffrax/misc/sde_kl_divergence.py

```diff
@@ -14,61 +13,77 @@ def _kl(drift1, drift2, diffusion):
-        y, _ = y
-        context = self.context(t)
-        aug_y = jnp.concatenate([y, context], axis=-1)
-        drift1 = self.drift1(t, aug_y, args)
-        drift2 = self.drift2(t, y, args)
-        diffusion = self.diffusion(t, y, args)
-        kl_divergence = jax.tree_map(_kl, drift1, drift2, diffusion)
-        kl_divergence = jax.tree_util.tree_reduce(operator.add, kl_divergence)
-        return drift1, kl_divergence
 
+class _AugControlTerm(ControlTerm):
+
+    control_term: ControlTerm
 
-class _AugDiffusion(eqx.Module):
-    diffusion: callable
+    def __init__(self, control_term: ControlTerm) -> None:
+        super().__init__(control_term.vector_field, control_term.control)
+        self.control_term = control_term
 
-    def __call__(self, t, y, args):
+    def vf(self, t: Scalar, y: PyTree, args: PyTree) -> PyTree:
         y, _ = y
-        diffusion = self.diffusion(t, y, args)
-        return diffusion, 0.0
+        vf = self.control_term.vf(t, y, args)
+        return vf, 0.0
 
+    def contr(self, t0: Scalar, t1: Scalar) -> PyTree:
+        return self.control_term.contr(t0, t1), 0.0
 
-class _AugBrownianPath(eqx.Module):
-    bm: AbstractBrownianPath
+    def vf_prod(self, t: Scalar, y: PyTree, args: PyTree, control: PyTree) -> PyTree:
+        y, _ = y
+        return self.control_term.vf_prod(t, y, args, control), 0.0
 
-    @property
-    def t0(self):
-        return self.bm.t0
 
-    @property
-    def t1(self):
-        return self.bm.t1
+class _AugVectorField(ODETerm):
 
-    def evaluate(self, t0, t1):
-        return self.bm.evaluate(t0, t1), 0.0
+    sde1: MultiTerm
+    sde2: MultiTerm
+    context: callable
+    kl: callable
+
+    def __init__(self, sde1, sde2, context) -> None:
+        super().__init__(sde1.terms[0].vector_field)
+        if sde1.terms[1] is not sde2.terms[1]:
+            raise ValueError("Two SDEs should share the same control terms")
+        self.sde1 = sde1
+        self.sde2 = sde2
+        if isinstance(self.sde1.terms[1], WeaklyDiagonalControlTerm):
+            self.kl = _kl_diagonal
+        else:
+            self.kl = _kl
+        self.context = context
+
+    def vf(self, t: Scalar, y: PyTree, args: PyTree) -> PyTree:
+        y, _ = y
+        context = self.context(t)
+        aug_y = y if context is None else jnp.concatenate([y, context], axis=-1)
+        drift1 = self.sde1.terms[0].vf(t, aug_y, args)
+        drift2 = self.sde2.terms[0].vf(t, y, args)
+        diffusion = self.sde1.terms[1].vf(t, y, args)
+        kl_divergence = jax.tree_map(self.kl, drift1, drift2, diffusion)
+        kl_divergence = jax.tree_util.tree_reduce(operator.add, kl_divergence)
+        return drift1, kl_divergence
 
 
 def sde_kl_divergence(
-    *,
-    drift1: callable,
-    drift2: callable,
-    diffusion: callable,
-    context: callable,
-    y0: PyTree,
-    bm: AbstractBrownianPath,
+    *, sde1: MultiTerm, sde2: MultiTerm, context: callable, y0: PyTree
```

Comment:
I've mentioned it earlier, but just to reiterate since this is the public API: I'd make this API accept `drift1: ODETerm, drift2: ODETerm, diffusion: AbstractTerm`, since that's what we actually need.

(e.g. a `MultiTerm` could include an arbitrary number of terms)